### DISCUSSION TITLE
High contrast mode

### DISCUSSION THREAD

**Discussion by yurua** [JunkFood02/Seal#364] [Q&A]
How can I make High Contrast Mode in XML, not Compose?
Thanks.

**Comment by JunkFood02** [comment]
Override `surface` and `background` to black in your dark theme should work

**Comment by yurua** [comment]
I'm using xml, not Compose, my app uses dynamic colours feature and dark
mode. I don't know how to change this 2 colours fun code..

пт, 25 нояб. 2022 г., 13:06 junkfood ***@***.***>:

**Comment by BobbyESP** [comment]
@yurua Hi, you have to add to your themes.xml some items. You can choose whatever you want to use.

This is the code you have to add:

```xml
    <style name="<Your app theme name>" parent="Theme.Material3.DayNight.NoActionBar">
        <item name="android:colorBackground">@color/high_contrast_background</item>
        <item name="android:colorForeground">@color/high_contrast_foreground</item>
        <item name="android:colorPrimary">@color/high_contrast_primary</item>
        <item name="android:colorPrimaryDark">@color/high_contrast_primary_dark</item>
        <item name="android:colorAccent">@color/high_contrast_accent</item>
    </style>
```

And you also have to add these items to the colors.xml

```xml
    <color name="high_contrast_background">#FF000000</color>
    <color name="high_contrast_foreground">#FFFFFFFF</color>
    <color name="high_contrast_primary">#FF000000</color>
    <color name="high_contrast_primary_dark">#FF000000</color>
    <color name="high_contrast_accent">#FF000000</color>
```
For example, if you just want to change the background color, you will only need to add the android:colorBackground to the themes.xml

You're free to use the ones that you want. If you have another question leave it here!

**Reply by yurua** [reply]
And last question)
How to activate this colors in code?
No compose!
Thanks.

**Reply by JunkFood02** [reply]
https://stackoverflow.com/questions/2482848/how-to-change-current-theme-at-runtime-in-android
